pkgs/build-support/gradle/patched-source.nix

```diff
@@ -0,0 +1,22 @@
+{ self
+, attrOverrides ? { }
+}:
+
+self.overrideAttrs (old: {
+  name = if old?pname then "${old.pname}-deps-${old.version}" else "${old.name}-deps";
```

Comment:
At least for nixpkgs I think we can rely on everything having pname+version especially going into the future.